### ISSUE TITLE
[1.12.2-rc1] endless loop when initializing swarm after daemon restart

### ISSUE THREAD

**Issue by cpuguy83** [docker/docker#27025]
**Steps to reproduce the issue:**
1. docker swarm init
2. \<add some things to swarm\>
3. restart the daemon

On step 2, this is going to be system dependent.
Basically there is a hard 5s timeout that needs to be triggered when initializing a swarm.
On faster disks it may be difficult to reproduce.

**Describe the results you received:**
Log is filled with:
```
DEBU[0011] (*Agent).UpdateTaskStatus                     module=agent task.id=bh5e8qgxw9zozsxpub9yct304
DEBU[0011] (*session).start                              module=agent
ERRO[0011] closing session after fatal error             error=rpc error: code = 3 desc = session invalid module=agent
ERRO[0011] failed reporting status to agent              error=rpc error: code = 3 desc = session invalid module=agent
```

Swarm is never started and this log is just repeated over and over, very quickly.

**Describe the results you expected:**
Swarm should initialize

**Comment by LK4D4** [comment]
@cpuguy83 I'm pretty sure there must be service to trigger this.

**Comment by cpuguy83** [comment]
lol, yes, step2 was being hidden because I had tags around it.

**Comment by tonistiigi** [comment]
@LK4D4 This was same as #26588 , right? So fixed.

**Comment by LK4D4** [comment]
@tonistiigi sure, thanks!

**Comment by thaJeztah** [comment]
@tonistiigi @LK4D4 I see https://github.com/docker/docker/issues/26588 was reopened, but is not on the 1.12.2 milestone - is this something to be fixed for 1.12.2?

**Comment by LK4D4** [comment]
@thaJeztah I think it's another bug, probably less important. That feature not even in 1.12 according to @tonistiigi
